iceoryx_hoofs/posix/time/source/deadline_timer.cpp

```diff
@@ -48,11 +51,18 @@ iox::units::Duration deadline_timer::remainingTime() const noexcept
     {
         return m_endTime - currentTime;
     }
-    return iox::units::Duration(std::chrono::milliseconds(0));
+    using namespace iox::units::duration_literals;
+    return 0_s;
 }
 
 iox::units::Duration deadline_timer::getCurrentMonotonicTime() noexcept
 {
-    return iox::units::Duration{std::chrono::steady_clock::now().time_since_epoch()};
+    timespec time_since_epoch{0, 0};
+    cxx::EnsuresWithMsg(!posix::posixCall(clock_gettime)(CLOCK_MONOTONIC, &time_since_epoch)
+                             .failureReturnValue(-1)
+                             .evaluate()
+                             .has_error(),
+                        "An error which should never happen occured during 'clock_gettime'!");
```

Comment:
Just in case you will be the reviewer when this is refactored with the new error handling. Please keep in mind to point out that `IOX_UNREACHABLE` shall be used :)